test/e2e/karmadactl_test.go

```diff
@@ -1311,6 +1311,203 @@ var _ = framework.SerialDescribe("Karmadactl taint testing", ginkgo.Labels{NeedC
+		karmadaCertSecret, err := kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Get(context.TODO(), "karmada-cert-secret", metav1.GetOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+		// Create a new secret named karmada-cert with the same data.
+		karmadaCert := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      options.KarmadaCertsName,
+				Namespace: util.NamespaceKarmadaSystem,
+			},
+			Data: karmadaCertSecret.Data,
+			Type: karmadaCertSecret.Type,
+		}
+
+		_, err = kubeHostClient.CoreV1().Secrets(util.NamespaceKarmadaSystem).Create(context.TODO(), karmadaCert, metav1.CreateOptions{})
+		gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			framework.RemoveSecret(kubeHostClient, util.NamespaceKarmadaSystem, options.KarmadaCertsName)
+		})
+	})
+
+	ginkgo.Context("Test disabling addons", func() {
+		ginkgo.It("should disable all addons except karmada-scheduler-estimator", func() {
+			ginkgo.By("Disabling all addons except karmada-scheduler-estimator", func() {
+				cmd := framework.NewKarmadactlCommand(kubeconfig, "", karmadactlPath, util.NamespaceKarmadaSystem, karmadactlTimeout*5, "addons", "disable", "all", "--karmada-kubeconfig", kubeconfig, "--context", karmadaHostContext, "--force")
+				_, err := cmd.ExecOrDie()
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			})
+
+			ginkgo.By("Verifying karmada-scheduler-estimator is still enabled on member clusters", func() {
+				cmd := framework.NewKarmadactlCommand(kubeconfig, "", karmadactlPath, util.NamespaceKarmadaSystem, karmadactlTimeout, "addons", "list", "-C", member1, "--karmada-kubeconfig", kubeconfig, "--context", karmadaHostContext)
+				output, err := cmd.ExecOrDie()
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				normalizedOutput := normalizeOutput(output)
+				checkExpectedAddons(normalizedOutput, map[string]string{"karmada-scheduler-estimator": "enabled"})
+			})
+
+			ginkgo.By("Ensuring karmada-scheduler-estimator is not enabled on Karmada Host", func() {
+				cmd := framework.NewKarmadactlCommand(kubeconfig, "", karmadactlPath, util.NamespaceKarmadaSystem, karmadactlTimeout, "addons", "list", "--karmada-kubeconfig", kubeconfig, "--context", karmadaHostContext)
+				output, err := cmd.ExecOrDie()
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				normalizedOutput := normalizeOutput(output)
+				checkExpectedAddons(normalizedOutput, map[string]string{"karmada-scheduler-estimator": "unknown"})
+			})
+		})
+	})
+
+	ginkgo.Context("Test enabling addons", func() {
+		ginkgo.It("should enable all addons except karmada-scheduler-estimator", func() {
+			ginkgo.By("Enabling all addons except karmada-scheduler-estimator", func() {
+				cmd := framework.NewKarmadactlCommand(
+					kubeconfig, "", karmadactlPath, util.NamespaceKarmadaSystem, karmadactlTimeout*5, "addons", "enable", "all", "--karmada-kubeconfig", kubeconfig, "--context", karmadaHostContext,
```

Comment:
```shell
# err log
I0810 19:26:08.768654  232078 check.go:42] Waiting for APIService(v1beta1.metrics.k8s.io) condition(Available), will try
```

Is the default 30-second timeout not enough? Try setting a larger flag `apiservice-timeout` value. If it still fails, there might be an issue.